### PR TITLE
Support for generating recursive structures [15346]

### PR DESCRIPTION
**Pull request by richiware**
According to OMG IDL standard, recursive structures are supported using `sequences` and forward declarations. This PR
add support for this and also:

- Improve test execution
- Fix bug generating bitset and bitmask

### CODE REVIEW

**Comment by MiguelCompany** [comment]
@richiware Would you mind resolving the conflicts here?

File: src/main/java/com/eprosima/idl/parser/typecode/SequenceTypeCode.java [hunk 1]
```diff
@@ -96,17 +106,20 @@ public String getIdlTypename()
...
     }
 
+    @Override
     public String getMaxsize()
     {
-        if(m_maxsize == null)
-            return "0";
+        if (m_maxsize == null)
+        {
+            return "100";
```
Comment (jsantiago-eProsima) [inline src/main/java/com/eprosima/idl/parser/typecode/SequenceTypeCode.java:114]: Why 100 and not 255 as stated in getMaxSerializedSize templates?

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]

File: src/main/java/com/eprosima/idl/parser/typecode/SequenceTypeCode.java (hunk 1 above)
Comment (richiware) [reply src/main/java/com/eprosima/idl/parser/typecode/SequenceTypeCode.java:114]: The value was always 100 for unbounded sequences. For example you can see this here:

https://github.com/eProsima/Fast-DDS-Gen/blob/master/src/main/java/com/eprosima/fastcdr/idl/templates/FastCdrCommon.stg#L360

File: src/main/java/com/eprosima/idl/context/Context.java [hunk 2]
```diff
@@ -17,22 +17,31 @@
...
 import com.eprosima.idl.parser.tree.Definition;
 import com.eprosima.idl.parser.tree.Interface;
 import com.eprosima.idl.parser.tree.Operation;
 import com.eprosima.idl.parser.tree.Param;
 import com.eprosima.idl.parser.tree.TypeDeclaration;
 import com.eprosima.idl.parser.tree.TreeNode;
+import com.eprosima.idl.parser.typecode.AliasTypeCode;
+import com.eprosima.idl.parser.typecode.AnyTypeCode;
+import com.eprosima.idl.parser.typecode.ArrayTypeCode;
+import com.eprosima.idl.parser.typecode.EnumTypeCode;
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/main/java/com/eprosima/idl/context/Context.java]: Done in 6f5e622

File: src/main/java/com/eprosima/idl/context/Context.java [hunk 3]
```diff
@@ -17,22 +17,31 @@
...
 import com.eprosima.idl.parser.typecode.EnumMember;
 import com.eprosima.idl.parser.typecode.EnumTypeCode;
 import com.eprosima.idl.parser.typecode.Kind;
+import com.eprosima.idl.parser.typecode.MapTypeCode;
 import com.eprosima.idl.parser.typecode.PrimitiveTypeCode;
+import com.eprosima.idl.parser.typecode.SequenceTypeCode;
+import com.eprosima.idl.parser.typecode.SetTypeCode;
+import com.eprosima.idl.parser.typecode.StringTypeCode;
 import com.eprosima.idl.parser.typecode.StructTypeCode;
+import com.eprosima.idl.parser.typecode.UnionTypeCode;
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/main/java/com/eprosima/idl/context/Context.java]: Done in 6f5e622

**Review by jsantiago-eProsima — APPROVED** [review]

**Pull request merged**